### PR TITLE
WIP-WG Create COMPONENT_GUIDANCE.md

### PR DESCRIPTION
**Pull request by sdalonzo**
WIP Guidance document on creating new components

### CODE REVIEW

File: COMPONENT_GUIDANCE.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+In the current front-end development enviroment, most projects are able to take advantage of tree-shaking. Tree-shaking is an optimization performed by a bundler (like webpack) to eliminate dead or unused code at build time. It is done by statically analyzing the source code and looking for exports in included packages that are not imported by the application.
+
+### Design System Upgrade Process
+
+There is an advantage to using a subpackage for a new component during the ongoing migration efforts from Design System v1 to v2. Subpackages can be used along side DSv1, like `icons` and `modal`. This means that the DS upgrade process need not block us from using awesome new components.
+
+## Recommendations
+
+Most new components added to the Design System should be subpackages. This allows users to use just what they need, which is a boon to performance. Consumers can start using the package when needed. The only exceptions might be deritives of existing core components, or other new primitives.
```
Comment (krevat) [inline COMPONENT_GUIDANCE.md]: deritives -> derivatives ?

**Comment by sdalonzo** [comment]
Add to `Topics of interest` in existing collaborator guide

**Comment by BeniCheni** [comment]
Sorry, @sdalonzo, I meant to say earlier that this content could be linked to the TOC over in [CONTRIBUTING guide](https://github.com/pricelinelabs/design-system/blob/master/CONTRIBUTING.md).

**Comment by sdalonzo** [comment]
@BeniCheni let's follow up and finish this up in the next week or two. With more folks getting involved in making DS contributions, it will be useful.

File: COMPONENT_GUIDANCE.md (hunk 1 above)
Comment (sdalonzo) [reply COMPONENT_GUIDANCE.md]: absolutely, thanks!

**Comment by sdalonzo** [comment]
declining in favor of #642

**Pull request closed without merging**